sql/moz-fx-data-shared-prod/telemetry_derived/newtab_interactions_v2/query.sql

```diff
@@ -0,0 +1,291 @@
+      event_name = "click"
+      AND mozfun.map.get_key(event_details, "is_follow_on") = "true"
+    ) AS follow_on_search_ad_clicks,
+    COUNTIF(
+      event_name = "click"
+      AND mozfun.map.get_key(event_details, "is_follow_on") = "true"
+      AND mozfun.map.get_key(event_details, "is_tagged") = "true"
+    ) AS tagged_follow_on_search_ad_clicks
+  FROM
+    events_unnested
+  WHERE
+    event_category IN ('newtab.search', 'newtab.search.ad')
+  GROUP BY
+    newtab_visit_id,
+    search_engine,
+    search_access_point
+),
+search_summary AS (
+  SELECT
+    newtab_visit_id,
+    ARRAY_AGG(
+      STRUCT(
+        search_engine,
+        search_access_point,
+        searches,
+        tagged_search_ad_clicks,
+        tagged_search_ad_impressions,
+        follow_on_search_ad_clicks,
+        follow_on_search_ad_impressions,
+        tagged_follow_on_search_ad_clicks,
+        tagged_follow_on_search_ad_impressions
+      )
+    ) AS search_interactions
+  FROM
+    search_events
+  GROUP BY
+    newtab_visit_id
+),
+topsites_events AS (
+  SELECT
+    mozfun.map.get_key(event_details, "newtab_visit_id") AS newtab_visit_id,
+    SAFE_CAST(mozfun.map.get_key(event_details, "position") AS INT64) AS topsite_position,
+    mozfun.map.get_key(event_details, "advertiser_name") AS topsite_advertiser_name,
+    mozfun.map.get_key(event_details, "tile_id") AS topsite_tile_id,
+    COUNTIF(event_name = 'impression') AS topsite_impressions,
+    COUNTIF(event_name = 'click') AS topsite_clicks,
+    COUNTIF(
+      event_name = 'impression'
+      AND mozfun.map.get_key(event_details, "is_sponsored") = "true"
+    ) AS sponsored_topsite_impressions,
+    COUNTIF(
+      event_name = 'impression'
+      AND mozfun.map.get_key(event_details, "is_sponsored") = "false"
+    ) AS organic_topsite_impressions,
+    COUNTIF(
+      event_name = 'click'
+      AND mozfun.map.get_key(event_details, "is_sponsored") = "true"
+    ) AS sponsored_topsite_clicks,
+    COUNTIF(
+      event_name = 'click'
+      AND mozfun.map.get_key(event_details, "is_sponsored") = "false"
+    ) AS organic_topsite_clicks,
+  FROM
+    events_unnested
+  WHERE
+    event_category = 'topsites'
+  GROUP BY
+    newtab_visit_id,
+    topsite_position,
+    topsite_advertiser_name,
+    topsite_tile_id
+),
+topsites_summary AS (
+  SELECT
+    newtab_visit_id,
+    ARRAY_AGG(
+      STRUCT(
+        topsite_advertiser_name,
+        topsite_position,
+        topsite_tile_id,
+        topsite_clicks,
+        sponsored_topsite_clicks,
+        organic_topsite_clicks,
+        topsite_impressions,
+        sponsored_topsite_impressions,
+        organic_topsite_impressions
+      )
+    ) AS topsite_interactions
+  FROM
+    topsites_events
+  GROUP BY
+    newtab_visit_id
+),
+pocket_events AS (
+  SELECT
+    mozfun.map.get_key(event_details, "newtab_visit_id") AS newtab_visit_id,
+    SAFE_CAST(mozfun.map.get_key(event_details, "position") AS INT64) AS pocket_story_position,
```

Comment:
I see the `SAFE_CAST`, do we want to filter out the `pocket_story_position IS NULL` values?